### PR TITLE
Update process for promoting a chart from incubator to stable

### PR DESCRIPTION
**Pull request by tompizmor**
**What this PR does / why we need it**:
It updates the documentation about how to promote a chart from incubator to stable.

**Special notes for your reviewer**:
I think that once a chart has been promoting from incubator to stable. The deprecated chart in the incubator repository should be removed to not have duplicates charts in the repo.

### CODE REVIEW

**Comment by tompizmor** [comment]
cc/ @scottrigby @viglesiasce

File: PROCESSES.md
Comment (unguiculus) [inline PROCESSES.md]: You should make it explicit that this has to be done in a separate PR.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: PROCESSES.md
Comment (tompizmor) [reply PROCESSES.md]: @unguiculus updated!

**Comment by prydonius** [comment]
/hold

IIRC this was an explicit decision because it prevents someone from re-creating a chart that was deprecating and accidentally overriding a previous version (e.g. if the chart got deprecated at 0.4.0, and a new chart started at 0.1.0, 0.4.0 would still be the latest).

Even if we did remove the chart from the incubator folder, it would still be in the chart repository (just marked as deprecated), so I'm not sure removing it here actually solves the duplication concern.

**Comment by tompizmor** [comment]
@prydonius I see the issue if you are just deprecating a chart from incubator. But if you are promoting from incubator to stable, I don't like to have the same chart in both folders. 
For example, the kubewatch chart has charts in incubator and stable. I don't expect that anyone will create a new chart for kubewatch in incubator if there is already one in stable no?

**Comment by prydonius** [comment]
@tompizmor okay, I see your point that it's a special case when moving from incubator to stable. That said, I still think removing it from the folder doesn't necessarily reduce the confusion since the chart will still be in both Helm repos. What do other maintainers think @kubernetes/charts-maintainers?

File: PROCESSES.md
Comment (mattfarina) [inline PROCESSES.md:46]: There are negative repercussions for existing applications that consume incubator charts as dependencies if this were to happen. Here are a couple examples:

1) Dependencies can be broken. If someone is referencing the chart in incubator in the [manner the helm docs describe](https://docs.helm.sh/developing_charts/#managing-dependencies-with-requirements-yaml) that other chart will then have a broken requirement because this one is gone.

2) If someone else comes along later and creates a new chart with the same name in incubator it can break apps. Another way to reference a chart as a requirement is by full url. If the old incubator chart and the new had the same name and version the new one would replace the old one. Apps consuming the old one at deploy time would now consume the new one and we would have introduced a silent error.

We don't want to change dependencies out from under people. This would be similar to the famous left-pad incident to do that.

**Review by mattfarina — CHANGES_REQUESTED** [review]

File: PROCESSES.md
Comment (prydonius) [reply PROCESSES.md:46]: > Dependencies can be broken. If someone is referencing the chart in incubator in the manner the helm docs describe that other chart will then have a broken requirement because this one is gone.

AFAIK this won't actually happen. The CI will not go an delete entries in the repo index for removed charts. Correct me if I'm wrong. The second concern is more the worry here.

Comment (unguiculus) [reply PROCESSES.md:46]: I thought I remember some discussion that deprecated charts should be removed, but maybe I'm wrong. I don't like deprecated charts to stay around forever. What if we create a file that tracks removal of deprecated charts? CI could then check this file when a new chart is added with the name of a deprecated one.

Comment (prydonius) [reply PROCESSES.md:46]: @unguiculus that could be interesting, shall we discuss it on tomorrows call?

Comment (davidkarlsen) [inline PROCESSES.md:46]: https://github.com/helm/charts/blob/master/README.md#repository-structure will need updating as well as it states "In order to get a Chart from incubator to stable, Chart maintainers should open a pull request that moves the chart folder." which indicates a plain move - not a delete/deprecate PR + a promote PR.

**Pull request closed without merging**
